src/lua/uri.lua

```diff
@@ -4,6 +4,19 @@ local ffi = require('ffi')
 local buffer = require('buffer')
 
 ffi.cdef[[
+struct uri_param {
+    const char *name;
+    int value_count;
+    const char **values;
+};
+
+/**
+ * We define all strings inside the `struct uri` as const, despite
+ * the fact that they are not constant in the C structure. This is
+ * necessary for the `uri_format` function work: there we cannot
+ * assign lua strings to a non-constant pointer. It's not a problem
+ * since uri_format doesn't change something in `struct uri`.
```

Comment:
This comment belongs to the previous commit.

Reply:
fixed